File: test_login.py
#!/usr/bin/env python3
import requests
import json

# Test login functionality
def test_login():
    url = "http://localhost:8001"
    
    # Test credentials  
    credentials = {
        'username': 'bbleze@outlook.com',
        'password': 'password123'
    }
    
    print(f"Testing login for: {credentials['username']}")
    
    # Send login request
    response = requests.post(
        f"{url}/auth/login",
        data=credentials,  # form data
        headers={'Content-Type': 'application/x-www-form-urlencoded'}
    )
    
    print(f"Login response status: {response.status_code}")
    print(f"Login response: {response.text}")
    
    if response.status_code == 200:
        token_data = response.json()
        access_token = token_data['access_token']
        print(f"Token obtained: {access_token[:50]}...")
        
        # Test getting profile with token
        profile_response = requests.get(
            f"{url}/auth/me",
            headers={'Authorization': f'Bearer {access_token}'}
        )
        
        print(f"Profile response status: {profile_response.status_code}")
        print(f"Profile response: {profile_response.text}")
        
        if profile_response.status_code == 200:
            user_data = profile_response.json()
            print(f"User role: {user_data.get('role')}")
            print("✅ Authentication flow working correctly!")
        else:
            print("❌ Profile fetch failed")
    else:
        print("❌ Login failed")

if __name__ == "__main__":
    test_login()

File: frontend/pages/api/test-login.ts
import { NextApiRequest, NextApiResponse } from 'next';

export default async function handler(req: NextApiRequest, res: NextApiResponse) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { email, password } = req.body;

    // Step 1: Login
    const loginResponse = await fetch('http://localhost:8001/auth/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: new URLSearchParams({
        username: email,
        password: password,
      }),
    });

    if (!loginResponse.ok) {
      const errorText = await loginResponse.text();
      return res.status(loginResponse.status).json({ 
        error: `Login failed: ${loginResponse.status} - ${errorText}` 
      });
    }

    const tokenData = await loginResponse.json();

    // Step 2: Get Profile
    const profileResponse = await fetch('http://localhost:8001/auth/me', {
      headers: {
        'Authorization': `Bearer ${tokenData.access_token}`,
      },
    });

    if (!profileResponse.ok) {
      const errorText = await profileResponse.text();
      return res.status(profileResponse.status).json({ 
        error: `Profile fetch failed: ${profileResponse.status} - ${errorText}` 
      });
    }

    const userData = await profileResponse.json();

    return res.status(200).json({
      success: true,
      token: tokenData.access_token,
      user: userData
    });

  } catch (error: any) {
    console.error('API Error:', error);
    return res.status(500).json({ 
      error: `Server error: ${error.message}` 
    });
  }
}